revolution_pi/gpio_pin.go

```diff
@@ -76,26 +132,52 @@ func (pin *gpioPin) Set(ctx context.Context, high bool, extra map[string]interfa
 
 // Get gets the high/low state of the pin.
 func (pin *gpioPin) Get(ctx context.Context, extra map[string]interface{}) (bool, error) {
-	pin.ControlChip.logger.Debugf("Reading from %v", pin.getGpioAddress())
-
 	if !pin.initialized {
 		return false, errors.New("pin not initialized")
 	}
-	return pin.ControlChip.getBitValue(int64(pin.getGpioAddress()), pin.BitPosition)
+	if pin.pwmMode {
+		return false, fmt.Errorf("cannot get pin state, Pin %s is configured as PWM", pin.Name)
+	}
+
+	gpioAddress := pin.getGpioAddress()
+	gpioBit := pin.BitPosition
+
+	if pin.isOutputPWM() {
```

Comment:
I thought we returned early on line 139 if this was a PWM pin. I wonder if this line is somehow wrong.

Reply:
this is the aliasing stuff, I put the comments within the if statement because I thought the code looked weird with them outside